frameworklauncher/src/main/java/com/microsoft/frameworklauncher/applicationmaster/RequestManager.java

```diff
@@ -406,6 +406,22 @@ public PlatformSpecificParametersDescriptor getPlatParams() {
     return taskResources;
   }
 
+  // Get a job's total Memory, CPU, and GPU request.  this interface will ignore the GPU attribute and Ports.
+  public ResourceDescriptor getJobTotalCountableResources() {
+    ResourceDescriptor totalResource = ResourceDescriptor.newInstance(0, 0, 0, (long) 0);
+    for (String taskRoleName : taskResources.keySet()) {
+      int taskNumber = taskRoles.get(taskRoleName).getTaskNumber();
```

Comment:
The method is not thread safe, for example, taskResources may not be consistent with taskRoles. Suggest to get a local taskRoles then only use it.